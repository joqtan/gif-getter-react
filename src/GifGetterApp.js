import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifGetterApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>Gif Getter</h2>
            <AddCategory setCategories={setCategories} />
            <hr />
            <ol>
                {categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))}
            </ol>
        </>
    )
}

export default GifGetterApp
