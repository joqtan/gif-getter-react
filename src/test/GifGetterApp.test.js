import { shallow } from 'enzyme'
import GifGetterApp from '../GifGetterApp'

describe('tests on <GifGetterApp/> component', () => {
    const wrapper = shallow(<GifGetterApp />)
    test('component should render properly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should show a list of categories', () => {
        const categories = ['star wars', 'lord of the ring']
        const wrapper = shallow(<GifGetterApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
})
