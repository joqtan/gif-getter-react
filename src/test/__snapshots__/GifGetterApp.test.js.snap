// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGetterApp/> component component should render properly 1`] = `
<Fragment>
  <h2>
    Gif Getter
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`tests on <GifGetterApp/> component should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Getter
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="star wars"
      key="star wars"
    />
    <GifGrid
      category="lord of the ring"
      key="lord of the ring"
    />
  </ol>
</Fragment>
`;
